src/wp-includes/widgets/class-wp-widget-archives.php

```diff
@@ -109,15 +109,31 @@ public function widget( $args, $instance ) {
 
 			<?php ob_start(); ?>
 <script>
-(function() {
-	var dropdown = document.getElementById( "<?php echo esc_js( $dropdown_id ); ?>" );
-	function onSelectChange() {
-		if ( dropdown.options[ dropdown.selectedIndex ].value !== '' ) {
-			document.location.href = this.options[ this.selectedIndex ].value;
-		}
+(function( dropdownId ) {
+	var dropdown = document.getElementById( dropdownId ), lastkey;
+	function onSelectChange(e) {
+		setTimeout(function() {
+			lastKey = dropdown.getAttribute( 'data-lastkey' );
```

Comment:
Note: The `lastkey` defined in the parent scope has a different case, so this `lastKey` would be set on `window`.

```suggestion
			const lastKey = dropdown.getAttribute( 'data-lastkey' );
```